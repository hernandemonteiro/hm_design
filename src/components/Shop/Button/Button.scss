.btn {
    width: 90%;
    padding: 3%;
    text-transform: uppercase;
    margin-bottom: 3%;
    border-radius: 10px;
    border: none;
    font-weight: bold;
    cursor: pointer;
}

.btn:hover {
    background-color: black;
    color: white;
}

.red {
    background-color: red;
    color: white;
}

.red:hover {
    background-color: black;
    color: white;
}