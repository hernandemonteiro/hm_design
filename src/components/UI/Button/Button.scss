.btn {
  width: 90%;
  padding: 3%;
  text-transform: uppercase;
  margin: 4%;
  border-radius: 10px;
  border: none;
  font-weight: bold;
  cursor: pointer;
  box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
}

.btn:hover {
  background-color: gainsboro;
}

.btn:disabled {
  background-color: rgba(173, 169, 169, 0.3);
  color: white;
  cursor: not-allowed
}

.transparent {
  background: none;
  color: yellow;
  box-shadow: none;
}

.transparent:hover:enabled {
  background: rgba(221, 221, 221, 0.1);
  color: yellow;
}

.red {
  background-color: red;
  color: white;
}

.red:hover:enabled {
  background-color: rgb(230, 1, 1);
  color: white;
}

.warning {
  background-color: orangered;
  color: white;
}

.warning:hover:enabled {
  background-color: rgb(245, 84, 26);
  color: white;
}

.green {
  background-color: green;
  color: white;
}

.green:hover:enabled {
  background-color: rgb(4, 179, 4);
  color: white;
}

.gradientHover:hover:enabled {
  background: rgb(2, 0, 36);
  background: linear-gradient(
    90deg,
    rgba(2, 0, 36, 1) 0%,
    rgba(107, 9, 121, 0.9472163865546218) 35%,
    rgba(0, 212, 255, 1) 100%
  );
  color: white;
}
