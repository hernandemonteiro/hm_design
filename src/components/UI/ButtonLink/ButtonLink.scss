.link {
    width: 100%;
    background-color: none;
}