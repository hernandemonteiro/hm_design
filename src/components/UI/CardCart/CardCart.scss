.cardCart {
    display: flex;
    justify-content: center;
    background-color: white;
    width: 100%;
    height: 110px;
    padding: 1%;
    margin: 2%;
    border-radius: 25px;
    text-align: center;
   
}

.photo {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 30%;
}

.description {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 60%;
    
}

.actions {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: flex-end;
    padding-right: 3%;
    padding-bottom: 4%;
    width: 10%;

}