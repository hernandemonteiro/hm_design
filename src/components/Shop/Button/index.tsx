import "./Button.scss";

interface ButtonProps {
  children: string;
  onClick?: any;
  className?: string;
}
export default function Button(props: ButtonProps) {
  return (
    <button className={`btn ${props.className}`} onClick={props.onClick}>
      {props.children}
    </button>
  );
}
