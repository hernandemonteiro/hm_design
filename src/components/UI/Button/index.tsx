import React from "react";
import "./Button.scss";

interface ButtonProps {
  children?: any;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  className?: string;
  type?: 'submit' | 'reset' | 'button';
  disabled?: boolean;
}
export default function Button(props: ButtonProps) {
  return (
    <button disabled={props.disabled} type={props.type} className={`btn ${props.className}`}  onClick={props.onClick}>
      {props.children}
    </button>
  );
}
