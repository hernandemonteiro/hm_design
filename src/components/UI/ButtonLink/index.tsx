import React from "react";
import {} from "../Button/Button.scss";
import {} from "./ButtonLink.scss";

interface buttonLinkProps {
  children?: any;
  className?: string;
  disabled?: boolean;
  to: string;
}
export default function ButtonLink(props: buttonLinkProps) {
  return (
    <a className="link" href={props.to}>
      <button
        className={`btn ${props.className}`}
        disabled={props.disabled}
        type="button"
      >
        {props.children}
      </button>
    </a>
  );
}
