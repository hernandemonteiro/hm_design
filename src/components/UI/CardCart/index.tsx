import React from "react";
import "./CardCart.scss";

interface CardProps {
  photo: string;
  description: string;
  actions: JSX.Element;
  price: string;
}
export default function CardCart(props: CardProps) {
  return (
    <div className="cardCart">
      <div className="photo">{props.photo}</div>
      <div className="description">{props.description}</div>
      <div className="actions">
        {props.actions}
        <h3>{props.price}</h3>
      </div>
    </div>
  );
}
