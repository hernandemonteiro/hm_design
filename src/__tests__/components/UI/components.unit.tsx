import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import Button from "../../../components/UI/Button";
import ButtonLink from "../../../components/UI/ButtonLink";
import CardCart from "../../../components/UI/CardCart";

describe("UI test", () => {
  it("<Button />", () => {
    const onClicked = jest.fn();
    render(<Button onClick={onClicked}>Test</Button>);
    const button = screen.getByRole("button");
    userEvent.click(button);
    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
    expect(onClicked).toHaveBeenCalledTimes(1);
  });

  it("<ButtonLink />", () => {
    const { container } = render(
      <ButtonLink to="/test">Test BtnLink</ButtonLink>
    );
    const buttonLink = screen.getByText("Test BtnLink");

    expect(container.getElementsByClassName("btn").length).toBe(1);
    expect(buttonLink).toBeInTheDocument();
    expect(buttonLink).toMatchSnapshot();
  });

  it("<CardCart />", () => {
    render(
      <CardCart
        photo={"photoTest"}
        description={"Test description"}
        actions={<Button>Action Test</Button>}
        price={"12.99"}
      />
    );

    const cardCartDescription = screen.getByText("Test description");
    expect(cardCartDescription).toBeInTheDocument();
    expect(cardCartDescription).toMatchSnapshot();
    const cardCartPrice = screen.getByText("12.99");
    expect(cardCartPrice).toBeInTheDocument();
    expect(cardCartPrice).toMatchSnapshot();
  });
});
