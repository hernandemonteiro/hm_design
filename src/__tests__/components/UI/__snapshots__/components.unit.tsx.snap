// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI test <Button /> 1`] = `
<button
  class="btn undefined"
>
  Test
</button>
`;

exports[`UI test <ButtonLink /> 1`] = `
<button
  class="btn undefined"
  type="button"
>
  Test BtnLink
</button>
`;

exports[`UI test <CardCart /> 1`] = `
<div
  class="description"
>
  Test description
</div>
`;

exports[`UI test <CardCart /> 2`] = `
<h3>
  12.99
</h3>
`;
